pages/api/trips/index.js

```diff
@@ -0,0 +1,31 @@
+import { getAllTrips, postTrip } from "../../../services/tripService";
+
+export default async function handler(req, res) {
+  const { method } = req;
+
+  switch (method) {
+    case "GET":
+      try {
+        const trips = await getAllTrips();
+        res.status(200).json(trips);
+      } catch (err) {
+        res.status(404).json({ error: "Not found" });
+      }
+      break;
+
+    case "POST":
+      try {
+        const newTrip = await postTrip(req.body);
+        if (!newTrip) return res.status(400).json({ error: "Invalid name" });
+        const trips = await getAllTrips();
+        res.status(201).json(trips);
```

Comment:
As long as http-method 'POST' means writing new Data to the database I am wondering, why do you 'getAllTrips' here?

Reply:
To keep one trustful source. Otherwise you can't be sure, that the data displayed on your client are the same as stored in the database.